lib/src/test/java/com/dynatrace/metric/util/NormalizeTest.java

```diff
@@ -261,4 +268,19 @@ static String repeatStringNTimes(String s, int n) {
             repeatStringNTimes("\\", 260),
             repeatStringNTimes("\\\\", 125)));
   }
+
+  private static Stream<Arguments> provideNeedToEscapeDimensionValue() {
+    return Stream.of(
+        Arguments.of("quote", "a\"b", true),
+        Arguments.of("space", "a b", true),
+        Arguments.of("comma", "a,b", true),
+        Arguments.of("equals", "a=b", true),
+        Arguments.of("backslash", "a\\b", true),
+        Arguments.of("need to escape multiple", "a\"b c,d=e\\f", true),
+        Arguments.of(
+            "no need to escape: a-Z",
+            "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ",
+            false),
+        Arguments.of("no need to escape: other", "_-", false));
```

Comment:
The empty string would also return false, right? Null would currently still fail with a nullpointer exception, see my comment about the null check above.